test/boskos.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+# acquires a project from boskos
+acquire_project() {
+    local project=""
+    local project_type="gce-project"
+
+    boskos_response=$(curl -X POST "http://boskos.test-pods.svc.cluster.local/acquire?type=${project_type}&state=free&dest=busy&owner=${JOB_NAME}")
+
+    if project=$(echo "${boskos_response}" | jq -r '.name'); then
+        echo "Using GCP project: ${project}"
+        CLOUDSDK_CORE_PROJECT="${project}"
+        export CLOUDSDK_CORE_PROJECT
+        heartbeat_project_forever &
+        BOSKOS_HEARTBEAT_PID=$!
+        export BOSKOS_HEARTBEAT_PID
+    else
+        (>&2 echo "ERROR: failed to acquire GCP project. boskos response was: ${boskos_response}")
+        exit 1
+    fi
+}
+
+# release the project back to boskos
+release_project() {
+    curl -X POST "http://boskos/release?name=${CLOUDSDK_CORE_PROJECT}&owner=${JOB_NAME}&dest=dirty"
+}
+
+# send a heartbeat to boskos for the project
+heartbeat_project() {
+    curl -X POST "http://boskos/update?name=${CLOUDSDK_CORE_PROJECT}&state=busy&owner=${JOB_NAME}" > /dev/null 2>&1
+}
+
+# heartbeat_project in an infinite loop
+heartbeat_project_forever() {
```

Comment:
i'm so glad we're open sourcing this :joy:

Reply:
/shrug